blog/content/post/2024-08-19-secureboot-freebsd.md

```diff
@@ -0,0 +1,378 @@
+
+1. Select `YES`
+
+1. Press `F10` to save and `Y` to confirm
+    ![f10 save dialog](/img/save_conf.png)
+
+1. Reset the platform by pressing `ESC` twice and selecting `Reset`
+
+#### Enrolling the ceritficate
+
+1. Press `F2` to enter setup again
+    ![setup menu](/img/setup_root.png)
+
+1. Navigate to `Device Manager/Secure Boot Configuration`
+    ![secure boot enabled](/img/sboot_enabled.png)
+
+1. Enable `Custom Mode` and enter the `Advanced Secure Boot Keys Management` 
+    menu again.
+    ![advanced secure boot keys management](/img/advanced_sboot.png)
+
+1. Navigate to `DB Options/Enroll Signature/Enroll Signature Using File/`.
+    If the EFI directory is properly mounted, you should see a single entry in
+    this menu, similar to this
+     ![efi directory entry](/img/efi_entry.png)
+
+1. Select the entry, and then select `testcert.cer`
+     ![testcert.cer](/img/testcert.png)
+
+1. Select `Commit Changes and Exit`
+     ![commit changes and exit](/img/commit_changes.png)
+
+1. Press `F10` to save and `Y` to confirm
+    ![f10 save dialog](/img/save_conf.png)
+
+1. Make sure that `Current Secure Boot State` is enabled
+    ![secure boot enabled](/img/sboot_enabled.png)
+
+1. Reset the platform again
+
+### Booting FreeBSD
+
+1. Wait for Dasharo to boot into UEFI Shell
+    ![EFI shell](/img/efi_shell.png)
+
+1. Enter the filesystem and boot the signed-loader.efi
+  ```
+  fs0:
+  signed-loader.efi
+  ```
+  ![EFI shell commands](/img/ush_commands.png)
+
+1. Say hello to Beastie
+    ![beastie](/img/beastie.png)
+
+
+## Testing on hardware
+
+Upon making sure that the loader-kernel object boots properly within an 
+emulation environment, we can proceed to Secure Booting FreeBSD on hardware.
+
+To do that, follow the exact same steps as with emulation, the only differewnce
+being that you will now need to upload the certificate to a USB drive and 
+enroll the certificate from there. 
+
+You will also need to place the `signed-loader.efi` file in your EFI partition,
+and add it as a custom boot option.
+
+### Adding a custom boot option
+
+* Enter setup menu
+* Navigate to `Boot Maintenance Manager/Boot Options/Add Boot Option`
+* Choose the appropriate disk label. They might look intimidating, but you
+  should be able to find the correct one by looking for a familiar keyword.
+  If you have an NVME drive for example, there should be an entry with 
+  `Pci(0x0,0x0)/NVMe`.
+* Find the loader-kernel object. It should be located under 
+  `<efi>/<freebsd>/loader.efi`
+* Name the entry appropriately, confirm and save the changes.
+
+
+## Troubleshooting
+
+### Mountroot
+
+If after booting the `signed-loader.efi` you land in `mountroot`, you need to
+type in `zfs:zroot/ROOT/default` and add the line
+
+```bash
+vfs.root.mountfrom="zfs:zroot/ROOT/default"
+```
+
+to your `/boot/loader.conf`.
+
+## Summary
+
+
+
+Unlock the full potential of your hardware and secure your firmware with the
```

Comment:
For the "marketing" part, would you like to propose something @pietrushnic ?
Perhaps mention the SB training as well?

Reply:
@pietrushnic please let me know if you wish to add something here, otherwise I can do it

Reply:
@macpijan yes, thanks for letting me know.

We should have something like that:

> If you want to deepen your understanding of UEFI Secure Boot and explore Intel Root of Trust technologies hands-on, consider joining our **DS08MSA: Mastering UEFI Secure Boot and Intel Root of Trust Technologies** training. This intensive course provides the operational knowledge and practical skills to work confidently with security technologies for x86 platforms. You'll learn to handle hardware assessments, configure UEFI Secure Boot, and provision Root of Trust for robust system security; for more details, visit our training page at [3mdeb Training](https://3mdeb.com/training/).


What is a dependency for that is an entry `DS08MSA: Mastering UEFI Secure Boot and Intel Root of Trust Technologies` based on https://hardwear.io/netherlands-2024/training/mastering-uefi-secure-boot-and-intel-root-of-trust-technologies.php

Reply:
[Added](https://github.com/3mdeb/news-and-ideas/pull/603/commits/469cc84541acf2428e46af01de19cf63b5bb0466#diff-1e142bc7519e97361fc0d3867abcb7fa3d257a4b5ccb79795e5e00acebf7d22fR393)